### ISSUE TITLE
Uncaught Error: Could not connect to X display

### ISSUE THREAD

**Issue by elimpnick** [atom/atom#15647]
[Enter steps to reproduce:]

1. Open any file associated with atom editor from Gnome
2. Type anything in the editor - crashes
3. Open atom from terminal, no crash.

**Atom**: 1.20.0 x64
**Electron**: 1.6.9
**OS**: linux 4.12.12-1-ARCH
**Thrown From**: Atom Core

### Stack Trace

Uncaught Error: Could not connect to X display.

```
At /usr/share/atom/resources/app/node_modules/keyboard-layout/lib/keyboard-layout.js:16

Error: Could not connect to X display.
    at get_manager (/usr/share/atom/resources/app/node_modules/keyboard-layout/lib/keyboard-layout.js:16:37)
    at getCurrentKeyboardLayout (/usr/share/atom/resources/app/node_modules/keyboard-layout/lib/keyboard-layout.js:32:17)
    at Object.getCurrentKeymap (/usr/share/atom/resources/app/node_modules/keyboard-layout/lib/keyboard-layout.js:22:31)
    at exports.keystrokeForKeyboardEvent (/usr/share/atom/resources/app/node_modules/atom-keymap/lib/helpers.js:244:66)
    at KeymapManager.module.exports.KeymapManager.keystrokeForKeyboardEvent (/usr/share/atom/resources/app/node_modules/atom-keymap/lib/keymap-manager.js:467:20)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app/node_modules/atom-keymap/lib/keymap-manager.js:347:30)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/usr/share/atom/resources/app/src/window-event-handler.js:100:42)
    at HTMLDocument.<anonymous> (/usr/share/atom/resources/app/src/window-event-handler.js:3:65)
```

### Commands

```
  2x -0:06.6.0 core:select-right (input.hidden-input)
     -0:06.1.0 intentions:highlight (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.30.5 
autocomplete-clang 0.11.3 
busy-signal 1.4.3 
git-plus 7.9.3 
intentions 1.1.5 
language-cmake 0.3.0 
linter 2.2.0 
linter-clang 4.1.2 
linter-ui-default 1.6.8 
platformio-ide-terminal 2.6.0 
```

**Comment by Ben3eeE** [comment]
@elimpnick Thank you for the report 🙇 What version of gnome are you using?

**Comment by elimpnick** [comment]
Gnome 3.22.0

**Comment by elimpnick** [comment]
Still happens on 1.21.0

**Comment by injcristianrojas** [comment]
I have the same issue. I'm using Fedora 27, GNOME 3.26.2, Atom 1.22.1 x64. In fact, I see the error window, and I can't even interact with it at all, because I get a modal saying "The editor has crashed".

**Comment by injcristianrojas** [comment]
As a workaround measure, I'm using the Flatpak version (1.23.1 on GNOME 3.26.2)and I have no issues.

**Comment by monteemc** [comment]
Had the same error. I was testing the use of 'screen' in bashrc. As soon I remove it, all worked again.
The problematic code in `.bashrc`: 
`
#Screen
#if [ -z "$STY" ]; then
#    exec screen -R
#fi
`

**Comment by injcristianrojas** [comment]
The behavior is quite strange. I have installed the atom-beta RPM package in Fedora, and, as @elimpnick says, when launching graphically, it crashes as descripted. But if you launch from command line, there are no problems.

**Comment by injcristianrojas** [comment]
I found something interesting: I've installed the RPM package, and I found that, when launching from the command line, the error is no more. And I found something even more interesting: When issuing a `which atom` on the command line, the system responds `/usr/bin/atom` whereas the launcher points to `/usr/share/atom/atom %F`. I modified the launcher to point to `atom %F` and the issue disappears.

Using 1.28.0 x64 on Fedora 28 (GNOME Shell 3.28.2) now.
